packages/vitest/src/node/plugins/index.ts

```diff
@@ -57,7 +57,7 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest('t
 
         ;(options as ResolvedConfig).defines = defines
 
-        let open: string | boolean | undefined
+        let open: string | boolean | undefined = false
```

Comment:
This doesn't actually fix the issue. Server still starts on the same port. What is needed here is to update `resolveApiConfig` method - it should treat `optimizer.web.enabled` and `optimizer.ssr.enabled` like `--api` flag. They are enabled by default, by the way, so they need to be explicitly set to false for us to not start a server.

This change hovewer should also be included in the fix. We only want to open a server when `ui` is enabled

Reply:
Okay, I've changed it, I'm not sure I got your point entirely

Reply:
Mostly yes, but I got a step further: #4095

What do you think?